### PR TITLE
Fixed issue  Unable to open URL for downloadable product

### PR DESCRIPTION
**Pull request by shikhamis11**
Fixed issue Unable to open URL for downloadable product #18944
Preconditions (*)

1.  Magento V2.2.6 Clean Install.
2.  PHP 7.0.28
3.  No special caching is present.

Steps to reproduce (*)

1.   Create downloadable product.
2.    Add a link to this product, for example https://google.com
3.    Save product.
4.    Order product in front-end.
5.    After order status complete navigate to your downloadable products in front-end.
6.   Click the button to open the URL set in step 2 to download your product.

**Expected result (*)**

    A new page of https://google.com is opened and shown in browser.

 **Actual result (*)**

    Page is not opened correctly and error by Magento is thrown:
    Something went wrong while getting the requested content.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by df2k2 — DISMISSED** [review]
Suggestions to simplify complexity?

```
    public function getContentType()
    {
        $this->_getHandle();
        if ($this->_linkType === self::LINK_TYPE_FILE) {
            if (function_exists('mime_content_type')
                && ($contentType = mime_content_type(
                    $this->_workingDirectory->getAbsolutePath($this->_resourceFile)
                ))
            ) {
                return $contentType;
            }
            return $this->_downloadableFile->getFileType($this->_resourceFile);
        }
        if ($this->_linkType === self::LINK_TYPE_URL) {
            return (is_array($this->_handle->stat($this->_resourceFile)['type'])
                ? end($this->_handle->stat($this->_resourceFile)['type'])
                : $this->_handle->stat($this->_resourceFile)['type']);
        }
        return $this->_contentType;
    }
```

**Review by davidverholen — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @davidverholen, thank you for the review.
[ENGCOM-3794](https://jira.corp.magento.com/browse/ENGCOM-3794) has been created to process this Pull Request

**Comment by sivaschenko** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, here is your [new Magento instance](https://pr-19996.instances.magento-community.engineering).
Admin access: [https://pr-19996.instances.magento-community.engineering/admin](https://pr-19996.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by sivaschenko** [comment]
Hi @shikhamis11 we found an issue on the PR code base, please take a look:

Manual testing scenario:

1. Create a downloadable product with a URL link type and value i.e. https://google.com
2. Create an order with the product and create an invoice for it
3. On Frontend Go to My Account -> My Downloadable Products
4. Click on the download link for the ordered product

"301 Moved" page is opened

**Comment by shikhamis11** [comment]
hello @sivaschenko thanks for the update 
It is fixed to show the actual result of the link set in the product earlier it was showing an error message.

**Comment by sivaschenko** [comment]
Hi @shikhamis11 , as I can see it is showing an error ("301 Moved") currently, on this PR code base.

**Comment by shikhamis11** [comment]
@sivaschenko I updated added a change please check that

**Comment by shikhamis11** [comment]
@sivaschenko with my commit there is error showing for codacy .. but the same piece of function is already being used in Magento .. can you suggest should I change all occurance og the get_hraders() function

**Comment by shikhamis11** [comment]
@sivaschenko any update on this?

**Comment by sivaschenko** [comment]
Hi @shikhamis11 thanks for the update. Is headers check applicable only for LINK_TYPE_URL?

**Comment by shikhamis11** [comment]
> Hi @shikhamis11 thanks for the update. Is headers check applicable only for LINK_TYPE_URL?

yes

**Comment by sivaschenko** [comment]
@shikhamis11 I suggest wrapping the headers check into the corresponding condition i.e. `if ($this->_linkType === self::LINK_TYPE_URL) {`

**Comment by shikhamis11** [comment]
@sivaschenko thanks for suggestion . I have updated code Please check now

File: app/code/Magento/Downloadable/Helper/Download.php [hunk 1]
```diff
@@ -252,10 +253,21 @@ public function setResource($resourceFile, $linkType = self::LINK_TYPE_FILE)
...
             }
         }
-
+        
         $this->_resourceFile = $resourceFile;
+        
+        /**
+        * check header for urls
+        */
+        if ($this->_linkType === self::LINK_TYPE_URL) {
```
Comment (sivaschenko) [inline app/code/Magento/Downloadable/Helper/Download.php]: @shikhamis11 please pay attention that `_linkType` property is set later in this method. I think it'll be better to compare to `$linkType`

**Review by sivaschenko — CHANGES_REQUESTED** [review]

File: app/code/Magento/Downloadable/Helper/Download.php (hunk 1 above)
Comment (shikhamis11) [reply app/code/Magento/Downloadable/Helper/Download.php]: Yes you are right I just updated with out noticing that let me correct this

**Comment by sdzhepa** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sdzhepa. Thank you for your request. I'm working on Magento instance for you

**Comment by sdzhepa** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sdzhepa. Thank you for your request. I'm working on Magento instance for you

**Comment by sivaschenko** [comment]
Hi @shikhamis11 during testing the PR we discovered that original issue cannot be reproduced on the latest 2.3-develop. Can you please recheck if the issue exists on the latest code base?

**Comment by shikhamis11** [comment]
sure

**Comment by shikhamis11** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11, here is your [Magento instance](https://i-19996-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-19996-2-3-develop.instances.magento-community.engineering/admin](https://i-19996-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by shikhamis11** [comment]
@sivaschenko no this is not working on current 2.3-develop instance . please check url 
screenshot_2019-02-08_17-53-14
screenshot_2019-02-08_17-55-10

**Comment by sivaschenko** [comment]
Sorry for confusion @shikhamis11 and thank you for the response! I rechecked and was able to reproduce it. Continuing to process the PR.

**Pull request merged**